storm-core/src/jvm/backtype/storm/Config.java

```diff
@@ -1697,19 +1697,45 @@
     public static final String TOPOLOGY_BOLTS_WINDOW_LENGTH_DURATION_MS = "topology.bolts.window.length.duration.ms";
 
     /*
-     * Bolt-specific configuration for windowed bolts to specifiy the sliding interval as a count of number of tuples.
+     * Bolt-specific configuration for windowed bolts to specify the sliding interval as a count of number of tuples.
      */
     @isInteger
     @isPositiveNumber
     public static final String TOPOLOGY_BOLTS_SLIDING_INTERVAL_COUNT = "topology.bolts.window.sliding.interval.count";
 
     /*
-     * Bolt-specific configuration for windowed bolts to specifiy the sliding interval in time duration.
+     * Bolt-specific configuration for windowed bolts to specify the sliding interval in time duration.
      */
     @isInteger
     @isPositiveNumber
     public static final String TOPOLOGY_BOLTS_SLIDING_INTERVAL_DURATION_MS = "topology.bolts.window.sliding.interval.duration.ms";
 
+    /*
+     * Bolt-specific configuration for windowed bolts to specify the name of the field in the tuple that holds
+     * the timestamp (e.g. the ts when the tuple was actually generated). If this config is specified and the
+     * field is not present in the incoming tuple, a java.lang.IllegalArgumentException will be thrown.
+     */
+    @isString
+    public static final String TOPOLOGY_BOLTS_TUPLE_TIMESTAMP_FIELD_NAME = "topology.bolts.tuple.timestamp.field.name";
```

Comment:
any reason for this config to have BOLT in them , can't we name it as topology.tuple.timestamp.field.name